inboxrecyclerview/src/main/java/me/saket/inboxrecyclerview/page/ExpandablePageLayout.kt

```diff
@@ -489,8 +538,9 @@ open class ExpandablePageLayout @JvmOverloads constructor(
     internalStateCallbacksForNestedPage.onPageAboutToExpand()
     internalStateCallbacksForRecyclerView.onPageAboutToExpand()
 
-    for (i in stateChangeCallbacks.indices.reversed()) {
-      stateChangeCallbacks[i].onPageAboutToExpand(expandAnimDuration)
+    // Reverse loop to let listeners remove themselves while in the loop.
+    for (i in stateChangeCallbacks.size - 1 downTo 0) {
```

Comment:
This would have the same issue as before since it's still just indexing based off the original size. You either need to back it with a `CopyOnWriteArrayList` + for-each, or make a defensive copy.

Reply:
Done: https://github.com/saket/InboxRecyclerView/pull/34/commits/6c5c7b7f81b11fe13ec0c8cea02ad1a1371a0487.